paddle/fluid/operators/mkldnn/conv_mkldnn_op.cc

```diff
@@ -220,9 +220,11 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
      * ('any') which lets a primitive (convolution in this case) choose
      * the memory format preferred for best performance
      */
-    std::string data_format = ctx.Attr<std::string>("data_format");
-    auto chosen_memory_format =
-        platform::data_format_to_memory_format(data_format);
+    // TODO(jczaja): This is workaround to make grad op UT's numerical
+    // gradient computation proper as this op is called directly without
+    // fetch op following it , so numercial grad is computed (in python)
+    // using block formats which will give wrong results
+    auto chosen_memory_format = is_test ? MKLDNNMemoryFormat::any : src_format;
```

Comment:
can you describe a bit of details about this workaround?

Reply:
@LeoZhao-Intel before this PR *data_format* was used in convolution to control which implementation will be executed by MKL-DNN eg. Any or NCHW. It could be left that way , but when NHWC is set to data_format in a model then performance goes drastically down.
To avoid performance degradation We no longer restrict MKL-DNN choice of implementation with *data_format* , but we use this information recognize what is the Paddle data format as it comes to model and what should be Paddle data format when MKL-DNN finish its work. 
Hence *choice_memory_format* is be default *any* to be the fastest. Why it is equal to src_format for training? It is because Grad UTs for mkl-dnn convolution are assuming that
convolution mkl-dnn op is computing FWD in NCHW. So as a temporary workaround
I put for training a *src_format* which for models with training is likely to be NCHW or NHWC.
This reduce performance of training. It is to be fixed once I have Unit tests for conv mkl-dnn GRAD working properly with Blocked layout.

@luotao1 I will restore performance of convolution in training & and revert TLS refactor in this PR